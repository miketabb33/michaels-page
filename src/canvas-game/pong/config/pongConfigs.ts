import { CanvasObject, CanvasObjectController } from '../../canvasObjectController'

export type PongConfig = {
  pongBall: CanvasObject
  playerPaddle: CanvasObject
  opponentPaddle: CanvasObject
  canvasDimensionUnits: number
  didFireFrame: (args: DidFireFrameArgs) => void
}

export type DidFireFrameArgs = {
  playerPaddle: CanvasObjectController
  pongBall: CanvasObjectController
  opponentPaddle: CanvasObjectController
  score: number
}
