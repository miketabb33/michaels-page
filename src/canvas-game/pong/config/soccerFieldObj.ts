import { colorTokens } from '../../../styles/colorTokens'
import { RenderableObject } from '../../render2dContext'

const horizontalLine = (canvasUnits: number): RenderableObject => {
  const height = 2
  return {
    rect: {
      position: {
        x: 0,
        y: canvasUnits / 2 - height / 2,
      },
      size: {
        width: canvasUnits,
        height,
      },
    },
    color: colorTokens.gray_50,
    shape: 'rectangle',
  }
}

const middleCircle = (canvasUnits: number): RenderableObject => {
  const size = 300
  return {
    rect: {
      position: {
        x: canvasUnits / 2 - size / 2,
        y: canvasUnits / 2 - size / 2,
      },
      size: {
        width: size,
        height: size,
      },
    },
    color: colorTokens.gray_50,
    shape: 'circle-outline',
  }
}

export const soccerField = (canvasDimensionUnits: number) => {
  return [horizontalLine(canvasDimensionUnits), middleCircle(canvasDimensionUnits)]
}
