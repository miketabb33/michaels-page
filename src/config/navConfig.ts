import { ENV } from './environments/currentEnv'

export type NavLinkConfig = {
  linkTo: string
  label: string
}

export const navConfig = (): NavLinkConfig[] => {
  const config = [
    { linkTo: '/', label: 'Home' },
    { linkTo: '/pong', label: 'Pong' },
  ]

  if (ENV.id === 'develop') return [...config, { linkTo: '/tic-tac-toe', label: 'T T T' }]
  return config
}
