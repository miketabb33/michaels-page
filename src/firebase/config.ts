import { initializeApp } from 'firebase/app'
import { getFirestore } from 'firebase/firestore'

const firebaseConfig = {
  apiKey: 'AIzaSyDprPwOCjEwEYcbW4nuPxFWtEOJFMc6bek',
  authDomain: 'michael-website-19ed0.firebaseapp.com',
  projectId: 'michael-website-19ed0',
  storageBucket: 'michael-website-19ed0.appspot.com',
  messagingSenderId: '855765427298',
  appId: '1:855765427298:web:763f3885091a33b5231d3b',
}

const app = initializeApp(firebaseConfig)
export const db = getFirestore(app)
