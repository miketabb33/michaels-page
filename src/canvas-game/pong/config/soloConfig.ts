import { random } from '../../../random'
import { Size } from '../../types/Size'
import { CanvasObject, CanvasObjectController } from '../../canvasObjectController'
import { ScoreResult } from '../../pongScore'
import { PongConfig } from './pongConfigs'
import { DirectionValue } from '../../types/DirectionValue'

const tuftsBlue = '#3C97D7'
const carrotOrange = '#D98F25'

export const getPongSoloConfig = (): PongConfig => {
  const canvasDimensionUnits = 1000
  return {
    pongBall: makePongBall(canvasDimensionUnits),
    playerPaddle: makePlayerPaddle(canvasDimensionUnits),
    opponentPaddle: makeOpponentPaddle(canvasDimensionUnits),
    canvasDimensionUnits,
    didFireFrame,
  }
}

const paddleSize: Size = { width: 150, height: 30 }

const makePlayerPaddle = (canvasUnits: number): CanvasObject => {
  return {
    rect: {
      size: paddleSize,
      position: {
        x: canvasUnits / 2 - paddleSize.width / 2,
        y: canvasUnits - paddleSize.height,
      },
    },
    color: tuftsBlue,
    velocity: {
      speed: 10,
      directionValue: {
        x: 0,
        y: 0,
      },
    },
    shape: 'rectangle',
  }
}

const makeOpponentPaddle = (canvasUnits: number): CanvasObject => {
  return {
    rect: {
      size: { ...paddleSize, width: canvasUnits },
      position: {
        x: 0,
        y: 0,
      },
    },
    color: tuftsBlue,
    velocity: {
      speed: 10,
      directionValue: {
        x: 0,
        y: 0,
      },
    },
    shape: 'rectangle',
  }
}

const makePongBall = (canvasUnits: number): CanvasObject => {
  const ballSize: Size = { width: 20, height: 20 }

  return {
    rect: {
      size: ballSize,
      position: {
        x: canvasUnits / 2 - ballSize.width / 2,
        y: canvasUnits / 2 - ballSize.height / 2 - 200,
      },
    },
    color: carrotOrange,
    velocity: {
      speed: levels[0].speed,
      directionValue: randomPaddleDirection(),
    },
    shape: 'circle',
  }
}

const randomPaddleDirection = (): DirectionValue => {
  const randomNum = random(6)

  if (randomNum === 0) return { x: 0.05, y: -0.95 }
  if (randomNum === 1) return { x: -0.5, y: -0.5 }
  if (randomNum === 2) return { x: 0.5, y: -0.5 }
  if (randomNum === 3) return { x: 0.05, y: 0.95 }
  if (randomNum === 4) return { x: -0.5, y: 0.5 }
  if (randomNum === 5) return { x: 0.5, y: 0.5 }
  return { x: 0, y: 0 }
}

const didFireFrame = (
  playerPaddle: CanvasObjectController,
  pongBall: CanvasObjectController,
  opponentPaddle: CanvasObjectController,
  scoreResult: ScoreResult
) => {
  const calculatedLevel = Math.floor(scoreResult.score / 4)
  const currentLevel = levels.find((pred) => pred.level === calculatedLevel) || levels[levels.length - 1]
  const currentBallSpeed = pongBall.canvasObj().velocity.speed

  if (currentBallSpeed !== currentLevel.speed) {
    pongBall.changeSpeed(currentLevel.speed)
    playerPaddle.changeSpeed(playerPaddle.canvasObj().velocity.speed - 0.5)
  }
}

type pongLevel = {
  speed: number
  level: number
}

const levels: pongLevel[] = [
  {
    level: 0,
    speed: 13,
  },
  {
    level: 1,
    speed: 14,
  },
  {
    level: 2,
    speed: 15,
  },
  {
    level: 3,
    speed: 16,
  },
  {
    level: 4,
    speed: 17,
  },
  {
    level: 5,
    speed: 18,
  },
  {
    level: 6,
    speed: 19,
  },
  {
    level: 7,
    speed: 20,
  },
]
