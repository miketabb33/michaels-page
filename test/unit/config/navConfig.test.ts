import { navConfig } from '../../../src/config/navConfig'

describe('Nav Config', () => {
  it('should match snapshot', () => {
    expect(navConfig).toMatchSnapshot()
  })
})
