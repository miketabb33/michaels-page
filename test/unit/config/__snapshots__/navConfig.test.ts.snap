// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Config should match snapshot 1`] = `[Function]`;
